### ISSUE TITLE
Generic chooser: links on result list lead to JSON response

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#11780]
### Issue Summary

Using the middle click on an entry in the result lists opens a JSON in a new tab.

It seems like the chooser modal uses this url to get its info on how to react on the selected result:
![Bildschirmfoto vom 2024-03-21 16-58-55](https://github.com/wagtail/wagtail/assets/468164/c3d199a4-5113-4d94-b292-034bfc6b3739)

Unfortunately, this leads some users to this page:
![Bildschirmfoto vom 2024-03-21 17-00-09](https://github.com/wagtail/wagtail/assets/468164/92263334-2445-4c8e-bcd3-815ddfb111b7)

To avoid having clueless users open a JSON page:
Can we use a different data-attribute to load the relevant data, but keep the edit-url in "href", so the middle click would work? Alternatively change the link to a button.

I've found https://github.com/wagtail/wagtail/issues/5408, which is basically the same issue with the page chooser, but in that case the link uses an anchor.

### Steps to Reproduce

1. Install the bakerydemo
2. Log into the CMS and use a model chooser, e.g. open http://127.0.0.1:8000/admin/pages/82/edit/#panel-child-content-authors-section and change the existing selection with "Choose another Person".

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.11
-   Django version: 4.2.x
-   Wagtail version: 5.2.x, latest main (6.1-dev)
-   Browser version: Firefox 123 (Linux)

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by lb-** [comment]
@th3hamm0r did the users right click and open in a new tab? How did they get the JSON response. The JavaScript is meant to make that link change the filtering within the modal.

**Comment by th3hamm0r** [comment]
@lb- Middle click (by clicking on the mouse wheel).
Yes, ctrl+click and click are handled by js, middle click bypasses it somehow.

**Comment by th3hamm0r** [comment]
Looks like there is a separate "auxclick"-event to handle those clicks:
https://developer.mozilla.org/en-US/docs/Web/API/Element/auxclick_event

**Comment by lb-** [comment]
Wow. Learned something new today, 

I guess technically these elements should be buttons which is the callout in #5408 

I remember now I did a PR to fix this actually https://github.com/wagtail/wagtail/pull/9209 but it wasn't merged and it got painful to rebase haha. 

The branch is still there though, maybe worth revisiting. We may also have some nicer options with Stimulus adoption though.

Maybe this bug can be closed and flagged as a duplicate of #5408 - and that issue updated to reflect Choosers generally.

**Comment by gasman** [comment]
I agree with @lb-'s assessment - this should be a button. I won't close this as a duplicate of #5408, as the page chooser works a bit differently from others (the data to be passed back to the calling page is stored on the link as `data-` attributes, to skip the roundtrip to the 'chosen' view that the other choosers perform) and would likely need to be implemented separately.
